tests/test_res1d.py

```diff
@@ -1,28 +1,108 @@
 import pytest
-from mikeio.res1d import Res1D, ExtractionPoint
 
+from mikeio.res1d import read, QueryData
 
-def get_test_query():
-    query = ExtractionPoint()
-    query.BranchName = "104l1"
-    query.Chainage = 34.4131
-    query.VariableType = "WaterLevel"
 
-    return query
+def test_query_validate():
+    # Good variable types
+    query = QueryData("WaterLevel")
+    query = QueryData("Discharge")
+    query = QueryData("Pollutant")
+
+    # Bad variable type
+    with pytest.raises(TypeError):
+        QueryData(666)
+
+    # Bad string variable type
+    with pytest.raises(ValueError):
+        QueryData("BadVariableType")
+
+    # Bad branch type
+    with pytest.raises(TypeError):
+        QueryData("WaterLevel", branch_name=666)
+
+    # Bad chainage type
+    with pytest.raises(TypeError):
+        QueryData("WaterLevel", "branch", chainage="BadChainage")
+
+    # Cannot set a chainage with no branch
+    with pytest.raises(ValueError):
+        QueryData("WaterLevel", None, 10)
+
+
+def test_query_repr():
+    query = QueryData("WaterLevel", "104l1", 34.4131)
+    expected = ("QueryData(variable_type='WaterLevel', branch_name='104l1', "
+                "chainage=34.4131)")
+    assert repr(query) == expected
+
+
+def test_query_iter():
+    query = QueryData("WaterLevel", "104l1", 34.4131)
+    vt, bn, c = query
+    assert vt == "WaterLevel"
+    assert bn == "104l1"
+    assert c == 34.4131
+
+
+@pytest.fixture
+def file():
+    return "tests/testdata/Exam6Base.res1d"
 
 
 def test_file_does_not_exist():
     file = "tests/testdata/not_a_file.res1d"
 
-    query = get_test_query()
+    query = QueryData("WaterLevel")
 
     with pytest.raises(FileExistsError):
-        assert Res1D().read(file, [query])
+        assert read(file, [query])
 
 
-def test_read_single_item():
-    file = "tests/testdata/Exam6Base.res1d"
-    query = get_test_query()
-    ts = Res1D().read(file, [query])
-
+@pytest.mark.parametrize("query,expected_max", [
+    (QueryData("WaterLevel", "104l1", 34.4131), 197.046),
+    (QueryData("WaterLevel", "9l1", 10), 195.165),
+    (QueryData("Discharge", "100l1", 23.8414), 0.1),
+    (QueryData("Discharge", "9l1", 5), 0.761)
+])
+def test_read_single_query(file, query, expected_max):
+    ts = read(file, [query])
     assert len(ts) == 110
+    assert pytest.approx(round(ts.max()[0], 3)) == expected_max
+
+
+def test_read_multiple_queries(file):
+    q1 = QueryData("WaterLevel", "104l1", 34.4131)
+    q2 = QueryData("Discharge", "9l1", 5)
+    ts = read(file, [q1, q2])
+    assert ts.shape == (110, 2)
+    ts_max = ts.max()
+    assert pytest.approx(round(ts_max[0], 3)) == 197.046
+    assert pytest.approx(round(ts_max[1], 3)) == 0.761
+
+
+def test_read_reach(file):
+    q_reach = QueryData("WaterLevel", "118l1")
+    ts = read(file, [q_reach])
```

Comment:
We should create a read function with a single query as input to avoid brackets

Reply:
Yes the read function could check the type of the query. I'd leave that for another PR if possible.

Reply:
Added it here: https://github.com/maximlt/mikeio/pull/2